src/clj/kefirnadar/application/async.clj

```diff
@@ -23,24 +20,25 @@
     conn
     {:tx-data (assemble-data-for-retraction data)}))
 
+
 (defn retract-old-ads-thread [running?]
-  "Check for ads older then predefined time, retrieve ids and feed them to a tx-retract-old-ads function."
-  (Thread.
-    ^Runnable
-    (while running?
-      (try
-        (let [conn (client/get-conn)
-              db (client/db)
-              last-month (Date/from (.minus (Instant/now) (Duration/ofDays 30)))
-              last-minute (Date/from (.minus (Instant/now) (Duration/ofMinutes 1)))
-              old-ad-ids (into [] (flatten (d/q '{:find  [(pull ?e [:db/id])]
-                                                  :in    [$ ?last-month]
-                                                  :where [[?e :ad/created ?created]
-                                                          [(< ?created ?last-month)]]}
-                                                db last-minute)))]
-          (if (empty? old-ad-ids)
-            (log/info "No ads older then 30 days to retract.")
-            (tx-retract-old-ads conn old-ad-ids)))
-        (catch Exception e
-          (log/info "Exception in retract-old-ads-thread:" e)))
-      (Thread/sleep 15000))))
+    "Check for ads older then predefined time, retrieve ids and feed them to a tx-retract-old-ads function."
+    (Thread.
+      ^Runnable
+      (while running?
+        (try
+          (let [conn (client/get-conn)
+                db (client/db)
+                last-month (Date/from (.minus (Instant/now) (Duration/ofDays 30)))
+                last-minute (Date/from (.minus (Instant/now) (Duration/ofMinutes 1)))
+                old-ad-ids (d/q '[:find (pull ?e pattern)
+                                             :in $ ?last-month pattern
```

Comment:
@radovanne3 query varijable treba da imaju prefiks `?`. Mislim da ovde nema potrebe da parametrizujemo pattern, pošto upit nije izvučen u var, tako da ga ne možemo koristiti opet na nekom drugom mestu. Ali dobra ideja da ga parametrizuješ.

Reply:
Mislio sam da je to standard pisanja neki ali okej, menjam i to. xD